Remove stale `cglobal(:jl_tls_states)` test

tkf opened the pull request:
It looks like the test

https://github.com/JuliaLang/julia/blob/22a192c197883b0e08df51e2e04b1ed35763d410/test/threads_exec.jl#L520-L527

is stale because there are no `jl_tls_states` anymore except in this test file:

```console
$ git grep '\bjl_tls_states\b'
test/threads_exec.jl:# `jl_tls_states` should only be defined on non-threading build.
test/threads_exec.jl:    cglobal(:jl_tls_states) != C_NULL
test/threads_exec.jl:    @test_throws ErrorException cglobal(:jl_tls_states)
```

Skipping the trivial patch #32284, it looks like the above test is 5-year old #17454.

It also looks like `jl_threading_enabled` can only return `1` nowadays. So I guess we can just delete this test?

cc @yuyichao

tkf committed:
Remove stale `cglobal(:jl_tls_states)` test

vtjnash reviewed (changes requested):
If we're removing the tests, we need to remove the jl_threading_enabled code too (which I agree is not useful and can be removed)

tkf committed:
Remove `jl_threading_enabled`

The function `threadingOn()` in `julia-config.jl` is the only reference
other than the tests just removed. It looks like `threadingOn` itself is
not used anywhere in the script.

tkf commented on contrib/julia-config.jl:
```diff
@@ -11,8 +11,6 @@ const options = [
     "--framework"
 ];
 
-threadingOn() = ccall(:jl_threading_enabled, Cint, ()) != 0
```
The function `threadingOn()` in `julia-config.jl` is the only use of `jl_threading_enabled` other than the tests just removed. It looks like `threadingOn` itself is not used anywhere in the script.
